python_modules/libraries/dagster-k8s/dagster_k8s/op_mutating_executor.py

```diff
@@ -0,0 +1,338 @@
+import logging
+from typing import Dict, Generic, Optional, TypeVar, Union, cast
+
+import dagster._seven as seven
+from dagster import (
+    DagsterEventType,
+    Field,
+    JsonMetadataValue,
+    _check as check,
+    executor,
+    usable_as_dagster_type,
+)
+from dagster._core.definitions.executor_definition import multiple_process_executor_requirements
+from dagster._core.events.log import EventLogEntry
+from dagster._core.execution.context.system import StepOrchestrationContext
+from dagster._core.execution.plan.handle import ResolvedFromDynamicStepHandle, StepHandle
+from dagster._core.execution.plan.inputs import (
+    FromPendingDynamicStepOutput,
+    FromStepOutput,
+    StepInput,
+)
+from dagster._core.execution.plan.outputs import StepOutputData
+from dagster._core.execution.retries import RetryMode
+from dagster._core.executor.base import Executor
+from dagster._core.executor.init import InitExecutorContext
+from dagster._core.executor.step_delegating import StepDelegatingExecutor, StepHandlerContext
+from dagster._core.storage.event_log import EventLogRecord, SqlEventLogStorage
+from dagster._core.storage.event_log.base import EventLogConnection, EventLogCursor
+from dagster._core.storage.event_log.schema import SqlEventLogStorageTable
+from dagster._core.storage.sqlalchemy_compat import db_select
+from dagster._serdes import deserialize_value
+from dagster._serdes.errors import DeserializationError
+from dagster._utils.merger import merge_dicts
+
+from dagster_k8s.container_context import K8sContainerContext
+from dagster_k8s.executor import _K8S_EXECUTOR_CONFIG_SCHEMA, K8sStepHandler
+from dagster_k8s.job import USER_DEFINED_K8S_CONFIG_KEY, UserDefinedDagsterK8sConfig
+from dagster_k8s.launcher import K8sRunLauncher
+
+_K8S_OP_EXECUTOR_CONFIG_SCHEMA = merge_dicts(
+    _K8S_EXECUTOR_CONFIG_SCHEMA,
+    {
+        "op_mutation_enabled": Field(
+            bool,
+            default_value=False,
+            description="enabled op configuration based on output of prior op",
+        )
+    },
+)
+
+
+USER_DEFINED_INPUT_K8S_OP_MUTATION_KEY = "dagster-k8s/mutation-enabled"
+
+T = TypeVar("T")
+
+
+@usable_as_dagster_type
+class K8sOpMutatingWrapper(Generic[T]):
+    value: T
+
+    def __init__(self, value: T):
+        self.value = value
+
+    @classmethod
+    def name(cls) -> str:
+        return cls.__name__
+
+
+def get_output_records_for_run_step(
```

Comment:
I don't think we can put storage logic like this in the shared dagster-k8s package that makes assumptions about specific tables being present in the storage - that responsibility would need to be delegated to the EventLogStorage class (for your own personal use it's fine to make that assumption, but since it's not part of the public API it could change in the future)